lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -753,24 +804,41 @@ class MongoClientInterface {
         });
     }
 
+    /**
+     * puts object metadata in bucket
+     * @param {String} bucketName bucket name
+     * @param {String} objName object name
+     * @param {Object} objVal object metadata
+     * @param {object} params params
+     * @param {Object} log logger
+     * @param {Function} cb callback
+     * @return {undefined}
+     */
     putObject(bucketName, objName, objVal, params, log, cb) {
         MongoUtils.serialize(objVal);
         const c = this.getCollection(bucketName);
-        if (params && params.versioning && !params.versionId) {
-            return this.putObjectVerCase1(c, bucketName, objName, objVal,
-                params, log, cb);
-        } else if (params && params.versionId === '') {
-            return this.putObjectVerCase2(c, bucketName, objName, objVal,
-                params, log, cb);
-        } else if (params && params.versionId && !params.repairMaster) {
-            return this.putObjectVerCase3(c, bucketName, objName, objVal,
-                params, log, cb);
-        } else if (params && params.versionId && params.repairMaster) {
-            return this.putObjectVerCase4(c, bucketName, objName, objVal,
-                params, log, cb);
-        }
-        return this.putObjectNoVer(c, bucketName, objName, objVal,
-            params, log, cb);
+        const _params = Object.assign({}, params);
```

Comment:
Consider `const _params = { ...params };` instead - no side-effect, no doc reading required, same cost in memory.

Reply:
Linter doesn't seem to like this one